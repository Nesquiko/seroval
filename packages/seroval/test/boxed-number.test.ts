import { describe, it, expect } from 'vitest';
import {
  crossSerialize,
  crossSerializeAsync,
  crossSerializeStream,
  serialize,
  serializeAsync,
  toCrossJSON,
  toCrossJSONAsync,
  toCrossJSONStream,
  toJSON,
  toJSONAsync,
} from '../src';

describe('boxed number', () => {
  describe('serialize', () => {
    it('supports boxed numbers', () => {
      const value = 0xDEADBEEF;
      expect(serialize(Object(value))).toBe(`Object(${value})`);
      expect(serialize(Object(NaN))).toBe('Object(0/0)');
      expect(serialize(Object(Infinity))).toBe('Object(1/0)');
      expect(serialize(Object(-Infinity))).toBe('Object(-1/0)');
      expect(serialize(Object(-0))).toBe('Object(-0)');
    });
  });
  describe('serializeAsync', () => {
    it('supports boxed numbers', async () => {
      const value = 0xDEADBEEF;
      expect(await serializeAsync(Promise.resolve(Object(value)))).toBe(`Promise.resolve(Object(${value}))`);
      expect(await serializeAsync(Promise.resolve(Object(NaN)))).toBe('Promise.resolve(Object(0/0))');
      expect(await serializeAsync(Promise.resolve(Object(Infinity)))).toBe('Promise.resolve(Object(1/0))');
      expect(await serializeAsync(Promise.resolve(Object(-Infinity)))).toBe('Promise.resolve(Object(-1/0))');
      expect(await serializeAsync(Promise.resolve(Object(-0)))).toBe('Promise.resolve(Object(-0))');
    });
  });
  describe('toJSON', () => {
    it('supports boxed numbers', () => {
      const value = 0xDEADBEEF;
      expect(JSON.stringify(toJSON(Object(value)))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Object(NaN)))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Object(Infinity)))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Object(-Infinity)))).toMatchSnapshot();
      expect(JSON.stringify(toJSON(Object(-0)))).toMatchSnapshot();
    });
  });
  describe('toJSONAsync', () => {
    it('supports boxed numbers', async () => {
      const value = 0xDEADBEEF;
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Object(value)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Object(NaN)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Object(Infinity)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Object(-Infinity)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toJSONAsync(Promise.resolve(Object(-0)))),
      ).toMatchSnapshot();
    });
  });
  describe('crossSerialize', () => {
    it('supports boxed numbers', () => {
      const value = 0xDEADBEEF;
      expect(crossSerialize(Object(value))).toMatchSnapshot();
      expect(crossSerialize(Object(NaN))).toMatchSnapshot();
      expect(crossSerialize(Object(Infinity))).toMatchSnapshot();
      expect(crossSerialize(Object(-Infinity))).toMatchSnapshot();
      expect(crossSerialize(Object(-0))).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports boxed numbers', () => {
        const value = 0xDEADBEEF;
        expect(crossSerialize(Object(value), { scopeId: 'example' })).toMatchSnapshot();
        expect(crossSerialize(Object(NaN), { scopeId: 'example' })).toMatchSnapshot();
        expect(crossSerialize(Object(Infinity), { scopeId: 'example' })).toMatchSnapshot();
        expect(crossSerialize(Object(-Infinity), { scopeId: 'example' })).toMatchSnapshot();
        expect(crossSerialize(Object(-0), { scopeId: 'example' })).toMatchSnapshot();
      });
    });
  });
  describe('crossSerializeAsync', () => {
    it('supports boxed numbers', async () => {
      const value = 0xDEADBEEF;
      expect(await crossSerializeAsync(Promise.resolve(Object(value)))).toMatchSnapshot();
      expect(await crossSerializeAsync(Promise.resolve(Object(NaN)))).toMatchSnapshot();
      expect(await crossSerializeAsync(Promise.resolve(Object(Infinity)))).toMatchSnapshot();
      expect(await crossSerializeAsync(Promise.resolve(Object(-Infinity)))).toMatchSnapshot();
      expect(await crossSerializeAsync(Promise.resolve(Object(-0)))).toMatchSnapshot();
    });
    describe('scoped', () => {
      it('supports boxed numbers', async () => {
        const value = 0xDEADBEEF;
        expect(await crossSerializeAsync(Promise.resolve(Object(value)), { scopeId: 'example' })).toMatchSnapshot();
        expect(await crossSerializeAsync(Promise.resolve(Object(NaN)), { scopeId: 'example' })).toMatchSnapshot();
        expect(await crossSerializeAsync(Promise.resolve(Object(Infinity)), { scopeId: 'example' })).toMatchSnapshot();
        expect(await crossSerializeAsync(Promise.resolve(Object(-Infinity)), { scopeId: 'example' })).toMatchSnapshot();
        expect(await crossSerializeAsync(Promise.resolve(Object(-0)), { scopeId: 'example' })).toMatchSnapshot();
      });
    });
  });

  describe('crossSerializeStream', () => {
    it('supports boxed numbers', async () => new Promise<void>((done) => {
      crossSerializeStream(Promise.resolve(Object(0xDEADBEEF)), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed NaN', async () => new Promise<void>((done) => {
      crossSerializeStream(Promise.resolve(Object(NaN)), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed Infinity', async () => new Promise<void>((done) => {
      crossSerializeStream(Promise.resolve(Object(Infinity)), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed -Infinity', async () => new Promise<void>((done) => {
      crossSerializeStream(Promise.resolve(Object(-Infinity)), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed -0', async () => new Promise<void>((done) => {
      crossSerializeStream(Promise.resolve(Object(-0)), {
        onSerialize(data) {
          expect(data).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    describe('scoped', () => {
      it('supports boxed numbers', async () => new Promise<void>((done) => {
        crossSerializeStream(Promise.resolve(Object(0xDEADBEEF)), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
      it('supports boxed NaN', async () => new Promise<void>((done) => {
        crossSerializeStream(Promise.resolve(Object(NaN)), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
      it('supports boxed Infinity', async () => new Promise<void>((done) => {
        crossSerializeStream(Promise.resolve(Object(Infinity)), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
      it('supports boxed -Infinity', async () => new Promise<void>((done) => {
        crossSerializeStream(Promise.resolve(Object(-Infinity)), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
      it('supports boxed -0', async () => new Promise<void>((done) => {
        crossSerializeStream(Promise.resolve(Object(-0)), {
          scopeId: 'example',
          onSerialize(data) {
            expect(data).toMatchSnapshot();
          },
          onDone() {
            done();
          },
        });
      }));
    });
  });
  describe('toCrossJSON', () => {
    it('supports boxed numbers', () => {
      const value = 0xDEADBEEF;
      expect(JSON.stringify(toCrossJSON(Object(value)))).toMatchSnapshot();
      expect(JSON.stringify(toCrossJSON(Object(NaN)))).toMatchSnapshot();
      expect(JSON.stringify(toCrossJSON(Object(Infinity)))).toMatchSnapshot();
      expect(JSON.stringify(toCrossJSON(Object(-Infinity)))).toMatchSnapshot();
      expect(JSON.stringify(toCrossJSON(Object(-0)))).toMatchSnapshot();
    });
  });
  describe('toCrossJSONAsync', () => {
    it('supports boxed numbers', async () => {
      const value = 0xDEADBEEF;
      expect(
        JSON.stringify(await toCrossJSONAsync(Promise.resolve(Object(value)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toCrossJSONAsync(Promise.resolve(Object(NaN)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toCrossJSONAsync(Promise.resolve(Object(Infinity)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toCrossJSONAsync(Promise.resolve(Object(-Infinity)))),
      ).toMatchSnapshot();
      expect(
        JSON.stringify(await toCrossJSONAsync(Promise.resolve(Object(-0)))),
      ).toMatchSnapshot();
    });
  });
  describe('toCrossJSONStream', () => {
    it('supports boxed numbers', async () => new Promise<void>((done) => {
      toCrossJSONStream(Promise.resolve(Object(0xDEADBEEF)), {
        onParse(data) {
          expect(JSON.stringify(data)).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed NaN', async () => new Promise<void>((done) => {
      toCrossJSONStream(Promise.resolve(Object(NaN)), {
        onParse(data) {
          expect(JSON.stringify(data)).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed Infinity', async () => new Promise<void>((done) => {
      toCrossJSONStream(Promise.resolve(Object(Infinity)), {
        onParse(data) {
          expect(JSON.stringify(data)).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed -Infinity', async () => new Promise<void>((done) => {
      toCrossJSONStream(Promise.resolve(Object(-Infinity)), {
        onParse(data) {
          expect(JSON.stringify(data)).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
    it('supports boxed -0', async () => new Promise<void>((done) => {
      toCrossJSONStream(Promise.resolve(Object(-0)), {
        onParse(data) {
          expect(JSON.stringify(data)).toMatchSnapshot();
        },
        onDone() {
          done();
        },
      });
    }));
  });
});
