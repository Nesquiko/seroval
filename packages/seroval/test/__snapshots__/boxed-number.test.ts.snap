// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`boxed number > crossSerialize > scoped > supports boxed numbers 1`] = `"($R=>$R[0]=Object(3735928559))($R["example"])"`;

exports[`boxed number > crossSerialize > scoped > supports boxed numbers 2`] = `"($R=>$R[0]=Object(0/0))($R["example"])"`;

exports[`boxed number > crossSerialize > scoped > supports boxed numbers 3`] = `"($R=>$R[0]=Object(1/0))($R["example"])"`;

exports[`boxed number > crossSerialize > scoped > supports boxed numbers 4`] = `"($R=>$R[0]=Object(-1/0))($R["example"])"`;

exports[`boxed number > crossSerialize > scoped > supports boxed numbers 5`] = `"($R=>$R[0]=Object(-0))($R["example"])"`;

exports[`boxed number > crossSerialize > supports boxed numbers 1`] = `"$R[0]=Object(3735928559)"`;

exports[`boxed number > crossSerialize > supports boxed numbers 2`] = `"$R[0]=Object(0/0)"`;

exports[`boxed number > crossSerialize > supports boxed numbers 3`] = `"$R[0]=Object(1/0)"`;

exports[`boxed number > crossSerialize > supports boxed numbers 4`] = `"$R[0]=Object(-1/0)"`;

exports[`boxed number > crossSerialize > supports boxed numbers 5`] = `"$R[0]=Object(-0)"`;

exports[`boxed number > crossSerializeAsync > scoped > supports boxed numbers 1`] = `"($R=>$R[0]=Promise.resolve($R[1]=Object(3735928559)))($R["example"])"`;

exports[`boxed number > crossSerializeAsync > scoped > supports boxed numbers 2`] = `"($R=>$R[0]=Promise.resolve($R[1]=Object(0/0)))($R["example"])"`;

exports[`boxed number > crossSerializeAsync > scoped > supports boxed numbers 3`] = `"($R=>$R[0]=Promise.resolve($R[1]=Object(1/0)))($R["example"])"`;

exports[`boxed number > crossSerializeAsync > scoped > supports boxed numbers 4`] = `"($R=>$R[0]=Promise.resolve($R[1]=Object(-1/0)))($R["example"])"`;

exports[`boxed number > crossSerializeAsync > scoped > supports boxed numbers 5`] = `"($R=>$R[0]=Promise.resolve($R[1]=Object(-0)))($R["example"])"`;

exports[`boxed number > crossSerializeAsync > supports boxed numbers 1`] = `"$R[0]=Promise.resolve($R[1]=Object(3735928559))"`;

exports[`boxed number > crossSerializeAsync > supports boxed numbers 2`] = `"$R[0]=Promise.resolve($R[1]=Object(0/0))"`;

exports[`boxed number > crossSerializeAsync > supports boxed numbers 3`] = `"$R[0]=Promise.resolve($R[1]=Object(1/0))"`;

exports[`boxed number > crossSerializeAsync > supports boxed numbers 4`] = `"$R[0]=Promise.resolve($R[1]=Object(-1/0))"`;

exports[`boxed number > crossSerializeAsync > supports boxed numbers 5`] = `"$R[0]=Promise.resolve($R[1]=Object(-0))"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed -0 1`] = `"($R=>$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed -0 2`] = `"($R=>($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(-0)))($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed -Infinity 1`] = `"($R=>$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed -Infinity 2`] = `"($R=>($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(-1/0)))($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed Infinity 1`] = `"($R=>$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed Infinity 2`] = `"($R=>($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(1/0)))($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed NaN 1`] = `"($R=>$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed NaN 2`] = `"($R=>($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(0/0)))($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed numbers 1`] = `"($R=>$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))())($R["example"])"`;

exports[`boxed number > crossSerializeStream > scoped > supports boxed numbers 2`] = `"($R=>($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(3735928559)))($R["example"])"`;

exports[`boxed number > crossSerializeStream > supports boxed -0 1`] = `"$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))()"`;

exports[`boxed number > crossSerializeStream > supports boxed -0 2`] = `"($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(-0))"`;

exports[`boxed number > crossSerializeStream > supports boxed -Infinity 1`] = `"$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))()"`;

exports[`boxed number > crossSerializeStream > supports boxed -Infinity 2`] = `"($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(-1/0))"`;

exports[`boxed number > crossSerializeStream > supports boxed Infinity 1`] = `"$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))()"`;

exports[`boxed number > crossSerializeStream > supports boxed Infinity 2`] = `"($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(1/0))"`;

exports[`boxed number > crossSerializeStream > supports boxed NaN 1`] = `"$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))()"`;

exports[`boxed number > crossSerializeStream > supports boxed NaN 2`] = `"($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(0/0))"`;

exports[`boxed number > crossSerializeStream > supports boxed numbers 1`] = `"$R[0]=($R[1]=(s,f,p)=>((p=new Promise((a,b)=>{s=a,f=b})).s=s,p.f=f,p))()"`;

exports[`boxed number > crossSerializeStream > supports boxed numbers 2`] = `"($R[3]=(p,d)=>{p.s(d),p.status="success",p.value=d;delete p.s;delete p.f})($R[0],$R[2]=Object(3735928559))"`;

exports[`boxed number > toCrossJSON > supports boxed numbers 1`] = `"{"t":21,"i":0,"f":{"t":0,"s":3735928559}}"`;

exports[`boxed number > toCrossJSON > supports boxed numbers 2`] = `"{"t":21,"i":0,"f":{"t":2,"s":7}}"`;

exports[`boxed number > toCrossJSON > supports boxed numbers 3`] = `"{"t":21,"i":0,"f":{"t":2,"s":5}}"`;

exports[`boxed number > toCrossJSON > supports boxed numbers 4`] = `"{"t":21,"i":0,"f":{"t":2,"s":6}}"`;

exports[`boxed number > toCrossJSON > supports boxed numbers 5`] = `"{"t":21,"i":0,"f":{"t":2,"s":4}}"`;

exports[`boxed number > toCrossJSONAsync > supports boxed numbers 1`] = `"{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":0,"s":3735928559}}}"`;

exports[`boxed number > toCrossJSONAsync > supports boxed numbers 2`] = `"{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":7}}}"`;

exports[`boxed number > toCrossJSONAsync > supports boxed numbers 3`] = `"{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":5}}}"`;

exports[`boxed number > toCrossJSONAsync > supports boxed numbers 4`] = `"{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":6}}}"`;

exports[`boxed number > toCrossJSONAsync > supports boxed numbers 5`] = `"{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":4}}}"`;

exports[`boxed number > toCrossJSONStream > supports boxed -0 1`] = `"{"t":22,"i":0,"f":{"t":26,"i":1,"s":1}}"`;

exports[`boxed number > toCrossJSONStream > supports boxed -0 2`] = `"{"t":23,"i":0,"a":[{"t":26,"i":3,"s":2},{"t":21,"i":2,"f":{"t":2,"s":4}}]}"`;

exports[`boxed number > toCrossJSONStream > supports boxed -Infinity 1`] = `"{"t":22,"i":0,"f":{"t":26,"i":1,"s":1}}"`;

exports[`boxed number > toCrossJSONStream > supports boxed -Infinity 2`] = `"{"t":23,"i":0,"a":[{"t":26,"i":3,"s":2},{"t":21,"i":2,"f":{"t":2,"s":6}}]}"`;

exports[`boxed number > toCrossJSONStream > supports boxed Infinity 1`] = `"{"t":22,"i":0,"f":{"t":26,"i":1,"s":1}}"`;

exports[`boxed number > toCrossJSONStream > supports boxed Infinity 2`] = `"{"t":23,"i":0,"a":[{"t":26,"i":3,"s":2},{"t":21,"i":2,"f":{"t":2,"s":5}}]}"`;

exports[`boxed number > toCrossJSONStream > supports boxed NaN 1`] = `"{"t":22,"i":0,"f":{"t":26,"i":1,"s":1}}"`;

exports[`boxed number > toCrossJSONStream > supports boxed NaN 2`] = `"{"t":23,"i":0,"a":[{"t":26,"i":3,"s":2},{"t":21,"i":2,"f":{"t":2,"s":7}}]}"`;

exports[`boxed number > toCrossJSONStream > supports boxed numbers 1`] = `"{"t":22,"i":0,"f":{"t":26,"i":1,"s":1}}"`;

exports[`boxed number > toCrossJSONStream > supports boxed numbers 2`] = `"{"t":23,"i":0,"a":[{"t":26,"i":3,"s":2},{"t":21,"i":2,"f":{"t":0,"s":3735928559}}]}"`;

exports[`boxed number > toJSON > supports boxed numbers 1`] = `"{"t":{"t":21,"i":0,"f":{"t":0,"s":3735928559}},"f":63,"m":[]}"`;

exports[`boxed number > toJSON > supports boxed numbers 2`] = `"{"t":{"t":21,"i":0,"f":{"t":2,"s":7}},"f":63,"m":[]}"`;

exports[`boxed number > toJSON > supports boxed numbers 3`] = `"{"t":{"t":21,"i":0,"f":{"t":2,"s":5}},"f":63,"m":[]}"`;

exports[`boxed number > toJSON > supports boxed numbers 4`] = `"{"t":{"t":21,"i":0,"f":{"t":2,"s":6}},"f":63,"m":[]}"`;

exports[`boxed number > toJSON > supports boxed numbers 5`] = `"{"t":{"t":21,"i":0,"f":{"t":2,"s":4}},"f":63,"m":[]}"`;

exports[`boxed number > toJSONAsync > supports boxed numbers 1`] = `"{"t":{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":0,"s":3735928559}}},"f":63,"m":[]}"`;

exports[`boxed number > toJSONAsync > supports boxed numbers 2`] = `"{"t":{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":7}}},"f":63,"m":[]}"`;

exports[`boxed number > toJSONAsync > supports boxed numbers 3`] = `"{"t":{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":5}}},"f":63,"m":[]}"`;

exports[`boxed number > toJSONAsync > supports boxed numbers 4`] = `"{"t":{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":6}}},"f":63,"m":[]}"`;

exports[`boxed number > toJSONAsync > supports boxed numbers 5`] = `"{"t":{"t":12,"i":0,"s":1,"f":{"t":21,"i":1,"f":{"t":2,"s":4}}},"f":63,"m":[]}"`;
